ee/app/livechat-enterprise/server/hooks/onMaxNumberSimultaneousChatsReached.js

```diff
@@ -0,0 +1,32 @@
+import { callbacks } from '../../../../../app/callbacks';
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { settings } from '../../../../../app/settings';
+import { LivechatInquiry } from '../../../../../app/models/server';
+import { checkWaitingQueue } from '../lib/Helper';
+
+callbacks.add('livechat.onMaxNumberSimultaneousChatsReached', (inquiry, agent) => {
+	if (!inquiry || !inquiry.defaultAgent) {
+		return inquiry;
+	}
+
+	if (!RoutingManager.getConfig().autoAssignAgent) {
+		return inquiry;
+	}
+
+	if (!settings.get('Livechat_last_chatted_agent_routing')) {
+		return inquiry;
+	}
+
+	const { _id, defaultAgent, department } = inquiry;
+
+	LivechatInquiry.removeDefaultAgentById(_id);
+
+	const { _id: defaultAgentId } = defaultAgent;
+	const { agentId } = agent;
+
+	if (defaultAgentId === agentId) {
+		checkWaitingQueue(department);
+	}
+
+	return LivechatInquiry.findOneById(_id);
```

Comment:
why is it returning a new `find` instead of just return the same input parameter `inquiry`?

Reply:
Because the `LivechatInquiry.removeDefaultAgentById(_id)` implemented above will change the state of the original object.